src/main/java/de/rub/nds/modifiablevariable/Logger/CustomLayout.java

```diff
@@ -0,0 +1,562 @@
+            String defaultPattern,
+            PatternSelector patternSelector,
+            boolean alwaysWriteExceptions,
+            boolean noConsoleNoAnsi) {
+        CustomLayout.SerializerBuilder builder = newSerializerBuilder();
+        builder.setAlwaysWriteExceptions(alwaysWriteExceptions);
+        builder.setConfiguration(configuration);
+        builder.setDefaultPattern(defaultPattern);
+        builder.setNoConsoleNoAnsi(noConsoleNoAnsi);
+        builder.setPattern(pattern);
+        builder.setPatternSelector(patternSelector);
+        builder.setReplace(replace);
+        return builder.build();
+    }
+
+    public String getConversionPattern() {
+        return this.conversionPattern;
+    }
+
+    public Map<String, String> getContentFormat() {
+        Map<String, String> result = new HashMap();
+        result.put("structured", "false");
+        result.put("formatType", "conversion");
+        result.put("format", this.conversionPattern);
+        return result;
+    }
+
+    public String toSerializable(LogEvent event) {
+        return this.eventSerializer.toSerializable(event);
+    }
+
+    public void serialize(LogEvent event, StringBuilder stringBuilder) {
+        this.eventSerializer.toSerializable(event, stringBuilder);
+    }
+
+    public void encode(LogEvent event, ByteBufferDestination destination) {
+        if (!(this.eventSerializer instanceof AbstractStringLayout.Serializer2)) {
+            super.encode(event, destination);
+        } else {
+            StringBuilder text =
+                    this.toText(
+                            (AbstractStringLayout.Serializer2) this.eventSerializer,
+                            event,
+                            getStringBuilder());
+            Encoder<StringBuilder> encoder = this.getStringBuilderEncoder();
+            encoder.encode(text, destination);
+            trimToMaxSize(text);
+        }
+    }
+
+    private StringBuilder toText(
+            AbstractStringLayout.Serializer2 serializer,
+            LogEvent event,
+            StringBuilder destination) {
+        return serializer.toSerializable(event, destination);
+    }
+
+    public static PatternParser createPatternParser(Configuration config) {
+        if (config == null) {
+            return new PatternParser(config, "Converter", LogEventPatternConverter.class);
+        } else {
+            PatternParser parser = (PatternParser) config.getComponent("Converter");
+            if (parser == null) {
+                parser = new PatternParser(config, "Converter", LogEventPatternConverter.class);
+                config.addComponent("Converter", parser);
+                parser = (PatternParser) config.getComponent("Converter");
+            }
+
+            return parser;
+        }
+    }
+
+    public String toString() {
+        return this.patternSelector == null
+                ? this.conversionPattern
+                : this.patternSelector.toString();
+    }
```

Comment:
This field overrides a super method and should therefore be annotated with @ Override (without the space). I suspect this being true for other methods in this class as well.